web/app/js/components/Navigation.test.jsx

```diff
@@ -122,3 +127,122 @@ describe('Navigation', () => {
     });
   });
 });
+
+describe('Namespace Select Button', () => {
+  it('displays All Namespaces as button text if the selected namespace is _all', () => {
+    const component = mount(
+      <BrowserRouter>
+        <Navigation
+          ChildComponent={() => null}
+          classes={{}}
+          theme={{}}
+          location={loc}
+          api={ApiHelpers("")}
+          releaseVersion="edge-1.2.3"
+          selectedNamespace="_all"
+          pathPrefix=""
+          uuid="fakeuuid" />
+      </BrowserRouter>
+    );
+
+    const button = component.find("Button");
+    expect(button).toIncludeText("All Namespaces");
+  });
+
+  it('renders emojivoto text', () => {
+    const component = mount(
+      <BrowserRouter>
+        <Navigation
+          ChildComponent={() => null}
+          classes={{}}
+          theme={{}}
+          location={loc}
+          api={ApiHelpers("")}
+          releaseVersion="edge-1.2.3"
+          selectedNamespace="emojivoto"
+          pathPrefix=""
+          uuid="fakeuuid" />
+      </BrowserRouter>
+    );
+
+    const button = component.find("Button");
+    expect(button).toIncludeText("emojivoto");
+  });
+
+  it('opens the Namespace Selection menu if button is clicked', () => {
+    const component = mount(
+      <BrowserRouter>
+        <Navigation
+          ChildComponent={() => null}
+          classes={{}}
+          theme={{}}
+          location={loc}
+          api={ApiHelpers("")}
+          releaseVersion="edge-1.2.3"
+          selectedNamespace="emojivoto"
+          pathPrefix=""
+          uuid="fakeuuid" />
+      </BrowserRouter>
+    );
+
+    expect(component.find("Menu").props().open).toBeFalsy();
+
+    const button = component.find("Button");
+    button.simulate("click");
+
+    expect(component.find("Menu").props().open).toBeTruthy();
+  });
+
+  describe('renders namespace selection menu with correct number of options', () => {
+    it('5 options', () => {
+      const component = mount(
+        <BrowserRouter>
+          <Navigation
+            ChildComponent={() => null}
+            classes={{}}
+            theme={{}}
+            location={loc}
+            api={ApiHelpers("")}
+            releaseVersion="edge-1.2.3"
+            selectedNamespace="emojivoto"
+            pathPrefix=""
+            uuid="fakeuuid" />
+        </BrowserRouter>
+      );
+
+      expect(component.find("Menu").find("MenuItem")).toHaveLength(2);
```

Comment:
Adding this for my own understanding -- It took me a second to remember that the Menu Items expected here are 2 -- "All Namespaces" and the "Select Namespace..." item. Then my understanding is you're adding the three namespaces from `namespaces` for a total of 5 menu items.

Reply:
I can add a comment explaining this if you think that it's necessary. Please, let me know.

Reply:
I think a comment would be great! It always helps when you come back to it later and have to remember what's going on